minio/resource_minio_iam_user.go

```diff
@@ -146,7 +155,7 @@ func minioUpdateUser(ctx context.Context, d *schema.ResourceData, meta interface
 		}
 	}
 
-	if iamUserConfig.MinioUpdateKey {
+	if iamUserConfig.MinioUpdateKey || d.HasChangeExcept(iamUserConfig.MinioSecret){
```

Comment:
What you want to put there, is actually `d.HasChange("secret")`, not `d.HasChangeExcept(iamUserConfig.MinioSecret)`